src/mainviews/create-job.tsx

```diff
@@ -195,6 +195,297 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
     // If no change in checkedness, don't do anything
   };
 
+  const parseTime = (input: string) => {
+    // Allow h:mm or hh:mm
+    const timeRegex = /^(\d\d?):(\d\d)$/;
+    const timeResult = timeRegex.exec(input);
+
+    let hours;
+    let minutes;
+
+    if (timeResult) {
+      hours = parseInt(timeResult[1]);
+      minutes = parseInt(timeResult[2]);
+    }
+
+    return [hours, minutes];
+  };
+
+  const validateTime = (input: string) => {
+    const errorMessage = trans.__('Time must be in hh:mm format');
+
+    const [hours, minutes] = parseTime(input);
+
+    if (
+      hours === undefined ||
+      minutes === undefined ||
+      hours < 0 ||
+      hours > 23 ||
+      minutes < 0 ||
+      minutes > 59
+    ) {
+      return errorMessage;
+    }
+
+    return '';
+  };
+
+  const handleScheduleTimeChange = (event: ChangeEvent) => {
```

Comment:
Typing event as `(event: React.ChangeEvent<HTMLInputElement>)` here and in other places would allow to avoid `(event.target as HTMLInputElement)` casting below